### PR TITLE
Add mixins that target the class as a comment.

### PR DESCRIPTION
**Pull request by modmuss50**
2am code, so prob got some issues. I plan on expanding this to add the comment to methods as well.

### CODE REVIEW

File: src/main/java/net/fabricmc/loom/util/MixinTargetScanner.java [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+	public void scan() {
+		classMixins = new HashMap<>();
+
+		Set<File> filesToScan = configuration.getResolvedConfiguration().getFiles();
+		filesToScan.forEach(this::scanFile);
+	}
+
+	private void scanFile(File file) {
+		//is this enough?
```
Comment (coderbot16) [inline src/main/java/net/fabricmc/loom/util/MixinTargetScanner.java]: Why not just check if it is a valid zip file?

**Comment by modmuss50** [comment]
I will most likely get this finished off without methods for now. Methods will take a bit more work

**Comment by modmuss50** [comment]
should be good now.

**Comment by asiekierka** [comment]
This would've been better implemented as an IDEA plugin or something, IMO. Decompilation of the Minecraft JAR was meant to be done once per version/mapping, not once per dependency change as this change seems to imply.

**Comment by modmuss50** [comment]
I would 1/2 agree with that, however it does help slove an issue people have been voicing for a while, without too much work.

As for it changing per dep, I do now see an issue with this (as JamiesWhiteShirt also pointed out), as if you have more than one project things will be fighting across them.

**Comment by modmuss50** [comment]
I think for now im going to revert it

**Pull request merged**